packages/worker/src/worker-options.ts

```diff
@@ -916,11 +968,43 @@ export function compileWorkerOptions(rawOpts: WorkerOptions, logger: Logger): Co
     defaultHeartbeatThrottleIntervalMs: msToNumber(opts.defaultHeartbeatThrottleInterval),
     loadedDataConverter: loadDataConverter(opts.dataConverter),
     activities,
+    nexusServices: compileNexusServices(opts),
     enableNonLocalActivities: opts.enableNonLocalActivities && activities.size > 0,
     tuner,
   };
 }
 
+function compileNexusServices(opts: WorkerOptions) {
+  const nexusServices = new Map<
+    string,
+    Map<string, nexus.OperationHandler<any, any> | nexus.SyncOperationHandler<any, any>>
+  >();
+  for (const s of opts.nexusServices ?? []) {
+    if (!s.name) {
+      throw new TypeError('Tried to register a Nexus service with no name');
+    }
+    if (nexusServices.has(s.name)) {
+      throw new TypeError(`Duplicate registration of nexus service ${s.name}`);
+    }
+    const ops = new Map<string, nexus.OperationHandler<any, any> | nexus.SyncOperationHandler<any, any>>();
```

Comment:
Should that be part of the Nexus SDK? Logic is trivial at present, but there I expect there will be some normalization work to be done at a later point (e.g. once we add decorators) which we wouldn't to diverge from one Nexus implementation to another. This also seems to be what we did in Java, and kind-of what we did in Go.

Reply:
Yeah, we discussed moving this to the Nexus SDK with Chad yesterday. I will do that.

Reply:
Oh, I missed that part of the conversation.

> I will do that.

I only wanted to confirm that we agree on direction. I can do it myself if you don't have time/prefer not to do it (same apply to all suggestions I make on this PR).